### PR TITLE
added GTMetrix API

### PR DESCRIPTION
**Pull request by ilyageekhub**
added basic UI

### CODE REVIEW

**Comment by ilyageekhub** [comment]
added GTMetrix API
added basic UI

File: PageSpeed/GTmetrix/GTMResaultViewController.swift [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+//  GTMResaultViewController.swift
+//  PageSpeed
+//
+//  Created by Admin on 19.02.2020.
+//  Copyright © 2020 Stanford University. All rights reserved.
+//
+
+import UIKit
+
+class GTMResaultViewController: UITableViewController {
```
Comment (nedopaka) [inline PageSpeed/GTmetrix/GTMResaultViewController.swift]: @ilyageekhub don't use shortcuts and avoid abbreviations like GTM in the project (see p.2, p.3 of [Development rules](https://github.com/ANScorp/pagespeed-swift/tree/develop#development-rules)). Rename `GTMResaultViewController` to **GTMetrixResaultViewController**.

File: PageSpeed/GTmetrix/GTMartixStructs.swift [hunk 2]
```diff
@@ -0,0 +1,100 @@
+//
+//  GTMartixStructs.swift
```
Comment (nedopaka) [inline PageSpeed/GTmetrix/GTMartixStructs.swift]: @ilyageekhub there is grammar error in the file name. Use **GTMetrix**, not `GTMatrix`.

File: PageSpeed/GTmetrix/GTMartixStructs.swift [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+//  PageSpeed
+//
+//  Created by Admin on 09.02.2020.
+//  Copyright © 2020 Stanford University. All rights reserved.
+//
+
+import Foundation
+
+struct GTMTestResponse: Decodable {
```
Comment (nedopaka) [inline PageSpeed/GTmetrix/GTMartixStructs.swift]: @ilyageekhub don't use shortcuts and avoid abbreviations like GTM in the project (see p.2, p.3 of [Development rules](https://github.com/ANScorp/pagespeed-swift/tree/develop#development-rules)). Use full name **GTMetrix** instead.

File: PageSpeed/GTmetrix/GTMatrixAPI.swift [hunk 4]
```diff
@@ -0,0 +1,56 @@
+//
+//  GTMatrixAPI.swift
```
Comment (nedopaka) [inline PageSpeed/GTmetrix/GTMatrixAPI.swift]: @ilyageekhub there is grammar error in the file name. Use **GTMetrix**, not `GTMatrix`.

File: PageSpeed/GTmetrix/GTMatrixAPI.swift [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+//  PageSpeed
+//
+//  Created by Admin on 09.02.2020.
+//  Copyright © 2020 Stanford University. All rights reserved.
+//
+
+import Moya
+
+enum GTMatrixAPI {
```
Comment (nedopaka) [inline PageSpeed/GTmetrix/GTMatrixAPI.swift]: @ilyageekhub there is grammar error in the name. Use **GTMetrix**, not `GTMatrix`.

File: PageSpeed/GTmetrix/GTMatrixAPI.swift [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    case testPage(url: String)
+    case testStatus(testID: String)
+}
+
+extension GTMatrixAPI: TargetType {
```
Comment (nedopaka) [inline PageSpeed/GTmetrix/GTMatrixAPI.swift]: @ilyageekhub there is grammar error in the name. Use **GTMetrix**, not `GTMatrix`.

File: PageSpeed/GTmetrix/Services/GTMTestService.swift [hunk 7]
```diff
@@ -0,0 +1,38 @@
+//
+//  GTMTestService.swift
```
Comment (nedopaka) [inline PageSpeed/GTmetrix/Services/GTMTestService.swift]: @ilyageekhub don't use shortcuts and avoid abbreviations like GTM in the project (see p.2, p.3 of [Development rules](https://github.com/ANScorp/pagespeed-swift/tree/develop#development-rules)). Use full name **GTMetrix** instead.

File: PageSpeed/GTmetrix/Services/GTMTestService.swift [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+//  PageSpeed
+//
+//  Created by Admin on 09.02.2020.
+//  Copyright © 2020 Stanford University. All rights reserved.
+//
+
+import Moya
+
+class GTMTestService: Service {
```
Comment (nedopaka) [inline PageSpeed/GTmetrix/Services/GTMTestService.swift]: @ilyageekhub don't use shortcuts and avoid abbreviations like GTM in the project (see p.2, p.3 of [Development rules](https://github.com/ANScorp/pagespeed-swift/tree/develop#development-rules)). Use full name **GTMetrix** instead.

File: PageSpeed/GTmetrix/Services/GTMTestStatusService.swift [hunk 9]
```diff
@@ -0,0 +1,38 @@
+//
+//  GTMTestService.swift
```
Comment (nedopaka) [inline PageSpeed/GTmetrix/Services/GTMTestStatusService.swift]: @ilyageekhub don't use shortcuts and avoid abbreviations like GTM in the project (see p.2, p.3 of [Development rules](https://github.com/ANScorp/pagespeed-swift/tree/develop#development-rules)). Use full name **GTMetrix** instead.

File: PageSpeed/GTmetrix/Services/GTMTestURLService.swift [hunk 10]
```diff
@@ -0,0 +1,68 @@
+//
+//  GTMTestURLService.swift
```
Comment (nedopaka) [inline PageSpeed/GTmetrix/Services/GTMTestURLService.swift]: @ilyageekhub don't use shortcuts and avoid abbreviations like GTM in the project (see p.2, p.3 of [Development rules](https://github.com/ANScorp/pagespeed-swift/tree/develop#development-rules)). Use full name **GTMetrix** instead.

File: PageSpeed/Test/NewTestViewController.swift [hunk 11]
```diff
@@ -20,7 +20,7 @@ class NewTestViewController: UIViewController {
     ]
     var mobilePageSpeedResult: PageSpeedResponse?
     var desktopPageSpeedResult: PageSpeedResponse?
-
+    var gtmResponse: GTMTestStatusResponse?
```
Comment (nedopaka) [inline PageSpeed/Test/NewTestViewController.swift]: @ilyageekhub use the same naming convention as above (e.g. **gtMetrixResult** for this case). Add a blank line between this declaration and enum.

File: PageSpeed/Test/NewTestViewController.swift [hunk 12]
```diff
@@ -50,9 +50,18 @@ class NewTestViewController: UIViewController {
     }
 
     // MARK: - Methods
-    func processRequestsToServices(url: String) {
+    func processRequestsToServices (url: String) {
```
Comment (nedopaka) [inline PageSpeed/Test/NewTestViewController.swift:53]: @ilyageekhub remove space after **processRequestsToServices**.

File: PageSpeed/Test/NewTestViewController.swift [hunk 13]
```diff
@@ -50,9 +50,18 @@ class NewTestViewController: UIViewController {
...
+        gtmResponse = nil
```
Comment (nedopaka) [inline PageSpeed/Test/NewTestViewController.swift]: @ilyageekhub is this line necessary? It should be removed.

File: PageSpeed/Test/NewTestViewController.swift [hunk 14]
```diff
@@ -50,9 +50,18 @@ class NewTestViewController: UIViewController {
...
-
+        dispatchGroup.enter()
+        GTMTestURLService(url: url).start { response, error in
+            if let error = error {
+                print(error)
+            } else if let response = response {
+                self.gtmResponse = response
+            }
+            dispatchGroup.leave()
+        }
```
Comment (nedopaka) [inline PageSpeed/Test/NewTestViewController.swift]: @ilyageekhub add a blank line before and after this block.

File: PageSpeed/Test/NewTestViewController.swift [hunk 15]
```diff
@@ -96,7 +105,7 @@ class NewTestViewController: UIViewController {
             testResultsViewController?.mobilePageSpeedResult = self.mobilePageSpeedResult
             testResultsViewController?.desktopPageSpeedResult = self.desktopPageSpeedResult
             testResultsViewController?.servicesArr = self.servicesArr
-
+            testResultsViewController?.gtmResponse = self.gtmResponse
```
Comment (nedopaka) [inline PageSpeed/Test/NewTestViewController.swift]: @ilyageekhub add a blank line after this.

File: PageSpeed/Test/NewTestViewController.swift [hunk 16]
```diff
@@ -140,7 +149,7 @@ class NewTestViewController: UIViewController {
             style: .cancel,
             handler: { ( _ : UIAlertAction) -> Void in
                 alertController.dismiss(animated: true, completion: dismissCompletion)
-            }
+        }
```
Comment (nedopaka) [inline PageSpeed/Test/NewTestViewController.swift]: @ilyageekhub irregular ident of the closure bracket.

File: PageSpeed/Test/TestResultsViewController.swift [hunk 17]
```diff
@@ -14,9 +14,9 @@ class TestResultsViewController: UIViewController {
     var url: String?
     var mobilePageSpeedResult, desktopPageSpeedResult: PageSpeedResponse?
     var servicesArr: [
-        (id: String, name: String)
+    (id: String, name: String)
```
Comment (nedopaka) [inline PageSpeed/Test/TestResultsViewController.swift]: @ilyageekhub irregular indent.

File: PageSpeed/Test/TestResultsViewController.swift [hunk 18]
```diff
@@ -14,9 +14,9 @@ class TestResultsViewController: UIViewController {
...
     ]?
-
+    var gtmResponse: GTMTestStatusResponse?
```
Comment (nedopaka) [inline PageSpeed/Test/TestResultsViewController.swift]: @ilyageekhub use the same naming convention as above (e.g. **gtMetrixResult** for this case). Add it before **servicesArr**.

File: PageSpeed/Test/TestResultsViewController.swift [hunk 19]
```diff
@@ -66,9 +66,31 @@ extension TestResultsViewController: UITableViewDataSource {
             }
             return "\(mobileScore) \(desktopScore)"
         case "gtmetrix":
-            return "Overall Score"
+            let score = gtmResponse?.results?.pageSpeedScore ?? 0
+            return "Page Speed Score: \(score)"
```
Comment (nedopaka) [inline PageSpeed/Test/TestResultsViewController.swift]: @ilyageekhub it's necessary to show **YSlow** score as well.

**Review by nedopaka — CHANGES_REQUESTED** [review]
@ilyageekhub check my comments.

**Comment by ANScorp** [comment]
@ilyageekhub please clarify what's the purpose of [PageSpeed/Errors/CustomError.swift](https://github.com/ANScorp/pagespeed-swift/pull/5/files#diff-cb059bf0f0ae8446810dda9d9d2dfa7c)? As I can see it's used at 53rd line of ** PageSpeed/GTmetrix/Services/GTMTestURLService.swift** only.

**Comment by nedopaka** [comment]
@ilyageekhub please clarify what's the purpose of [PageSpeed/Errors/CustomError.swift](https://github.com/ANScorp/pagespeed-swift/pull/5/files#diff-cb059bf0f0ae8446810dda9d9d2dfa7c)? As I can see it's used at 53rd line of **PageSpeed/GTmetrix/Services/GTMTestURLService.swift** only.

**Comment by ilyageekhub** [comment]
all fixed

File: PageSpeed/GTmetrix/GTMertixStructs.swift [hunk 20]
```diff
@@ -1,5 +1,5 @@
 //
-//  GTMartixStructs.swift
+//  GTMertixStructs.swift
```
Comment (nedopaka) [inline PageSpeed/GTmetrix/GTMertixStructs.swift:2]: @ilyageekhub there is grammar error in the file name. Use **GTMetrix**, not `GTMertix`.

File: PageSpeed/GTmetrix/GTMetrixResaultViewController.swift [hunk 21]
```diff
@@ -8,7 +8,7 @@
 
 import UIKit
 
-class GTMResaultViewController: UITableViewController {
+class GTMetrixResaultViewController: UITableViewController {
```
Comment (nedopaka) [inline PageSpeed/GTmetrix/GTMetrixResaultViewController.swift]: @ilyageekhub there is grammar error in the file name. Use **GTMetrixResultViewController**, not `GTMetrixResaultViewController`.

File: PageSpeed/GTmetrix/GTMertixStructs.swift [hunk 22]
```diff
@@ -68,14 +68,14 @@ enum TestState: String {
     case queued, started, completed, error
 }
 
-struct GTMTestStatusResponse: Decodable {
+struct GTMetrixTestStatusResponse: Decodable {
```
Comment (nedopaka) [inline PageSpeed/GTmetrix/GTMertixStructs.swift]: @ilyageekhub why so weird name for struct that is referring to response from GTMetrix service. Should be **GTMetrixResponse**. Please, check structs names.
According to p.3
>Classes and structures should have informative and explicitly defined names.

File: PageSpeed/GTmetrix/Services/GTMetrixTestStatusService.swift [hunk 23]
```diff
@@ -8,23 +8,23 @@
 
 import Moya
 
-class GTMTestStatusService: Service {
+class GTMetrixTestStatusService: Service {
```
Comment (nedopaka) [inline PageSpeed/GTmetrix/Services/GTMetrixTestStatusService.swift]: @ilyageekhub why so weird name for class that is responsible for making request to GTMetrix? Please, check classes names.
According to p.3
>Classes and structures should have informative and explicitly defined names.

File: PageSpeed/Test/NewTestViewController.swift [hunk 24]
```diff
@@ -20,7 +20,7 @@ class NewTestViewController: UIViewController {
     ]
     var mobilePageSpeedResult: PageSpeedResponse?
     var desktopPageSpeedResult: PageSpeedResponse?
-    var gtmResponse: GTMTestStatusResponse?
+    var gtMetrixResponse: GTMetrixTestStatusResponse?
```
Comment (nedopaka) [inline PageSpeed/Test/NewTestViewController.swift]: @ilyageekhub as for `GTMetrixTestStatusResponse` again, check names for structs and classes.
>Classes and structures should have informative and explicitly defined names.

File: PageSpeed/Test/NewTestViewController.swift (hunk 12 above)
Comment (nedopaka) [reply PageSpeed/Test/NewTestViewController.swift:53]: @ilyageekhub once again, it's not fixed.

File: PageSpeed/Test/NewTestViewController.swift [hunk 25]
```diff
@@ -77,7 +77,6 @@ class NewTestViewController: UIViewController {
                 dispatchGroup.leave()
             }
         }
-
```
Comment (nedopaka) [inline PageSpeed/Test/NewTestViewController.swift:71]: @ilyageekhub why you removed blank line? Here should be blank line according to p.2. See https://github.com/raywenderlich/swift-style-guide#spacing.

File: PageSpeed/Test/NewTestViewController.swift [hunk 26]
```diff
@@ -93,19 +92,18 @@ class NewTestViewController: UIViewController {
...
```
Comment (nedopaka) [inline PageSpeed/Test/NewTestViewController.swift:87]: @ilyageekhub why you removed blank line? Here should be blank line according to p.2. See https://github.com/raywenderlich/swift-style-guide#spacing.

File: PageSpeed/Test/NewTestViewController.swift [hunk 27]
```diff
@@ -93,19 +92,18 @@ class NewTestViewController: UIViewController {
...
         dispatchGroup.notify(queue: .main) {
             let testResultsViewController = UIStoryboard(
                 name: "Stage-A",
                 bundle: nil
             ).instantiateViewController(identifier: "TestResultsViewController")
                 as? TestResultsViewController
-
```
Comment (nedopaka) [inline PageSpeed/Test/NewTestViewController.swift:94]: @ilyageekhub why you removed blank line? Here should be blank line according to p.2. See https://github.com/raywenderlich/swift-style-guide#spacing.

File: PageSpeed/Test/NewTestViewController.swift [hunk 28]
```diff
@@ -93,19 +92,18 @@ class NewTestViewController: UIViewController {
...
             testResultsViewController?.url = self.urlTextField.text
             testResultsViewController?.mobilePageSpeedResult = self.mobilePageSpeedResult
             testResultsViewController?.desktopPageSpeedResult = self.desktopPageSpeedResult
             testResultsViewController?.servicesArr = self.servicesArr
-            testResultsViewController?.gtmResponse = self.gtmResponse
+
```
Comment (nedopaka) [inline PageSpeed/Test/NewTestViewController.swift]: @ilyageekhub remove this line.

File: PageSpeed/Test/NewTestViewController.swift [hunk 29]
```diff
@@ -93,19 +92,18 @@ class NewTestViewController: UIViewController {
...
+            testResultsViewController?.gtmetrixResponse = self.gtMetrixResponse
```
Comment (nedopaka) [inline PageSpeed/Test/NewTestViewController.swift]: @ilyageekhub variables and properties should be in **camelCase**. See https://github.com/raywenderlich/swift-style-guide#naming.

File: PageSpeed/Test/NewTestViewController.swift [hunk 30]
```diff
@@ -138,7 +136,9 @@ class NewTestViewController: UIViewController {
         }
     }
 
-    func showAlert(title: String, message: String, dismissCompletion: (() -> Void)? = nil) {
+    func showAlert(title: String,
+                   message: String,
+                   dismissCompletion: (() -> Void)? = nil) {
```
Comment (nedopaka) [inline PageSpeed/Test/NewTestViewController.swift]: @ilyageekhub redundant new lines for input parameters. Don't change this block.

File: PageSpeed/Test/TestResultsViewController.swift (hunk 17 above)
Comment (nedopaka) [reply PageSpeed/Test/TestResultsViewController.swift]: @ilyageekhub once again, it's not fixed.

File: PageSpeed/Test/TestResultsViewController.swift [hunk 31]
```diff
@@ -16,7 +16,7 @@ class TestResultsViewController: UIViewController {
     var servicesArr: [
     (id: String, name: String)
     ]?
-    var gtmResponse: GTMTestStatusResponse?
+    var gtmetrixResponse: GTMetrixTestStatusResponse?
```
Comment (nedopaka) [inline PageSpeed/Test/TestResultsViewController.swift]: @ilyageekhub variables and properties should be in **camelCase**. See https://github.com/raywenderlich/swift-style-guide#naming.
Also, check names for structs and classes. Names should be consistent and make sense.
>Classes and structures should have informative and explicitly defined names.

**Review by nedopaka — CHANGES_REQUESTED** [review]
@ilyageekhub fix grammar errors. Also, there is some mess with naming. Check names for structs and classes. Names should be consistent and make sense. Variables and properties should be in **camelCase**.
Also, why don't you **MARK** your code?

File: PageSpeed/Test/TestResultsViewController.swift (hunk 31 above)
Comment (ilyageekhub) [reply PageSpeed/Test/TestResultsViewController.swift]: done

File: PageSpeed/Test/NewTestViewController.swift (hunk 30 above)
Comment (ilyageekhub) [reply PageSpeed/Test/NewTestViewController.swift]: done

File: PageSpeed/Test/NewTestViewController.swift (hunk 29 above)
Comment (ilyageekhub) [reply PageSpeed/Test/NewTestViewController.swift]: done

File: PageSpeed/Test/TestResultsViewController.swift (hunk 17 above)
Comment (ilyageekhub) [reply PageSpeed/Test/TestResultsViewController.swift]: what do you mean behind "irregular indent"?

File: PageSpeed/GTmetrix/GTMetrixResaultViewController.swift (hunk 21 above)
Comment (ilyageekhub) [reply PageSpeed/GTmetrix/GTMetrixResaultViewController.swift]: done

File: PageSpeed/GTmetrix/GTMetrixResultViewController.swift [hunk 32]
```diff
@@ -0,0 +1,173 @@
+//
+//  GTMetrixResaultViewController.swift
```
Comment (nedopaka) [inline PageSpeed/GTmetrix/GTMetrixResultViewController.swift:2]: @ilyageekhub this comment line should match file name.

File: PageSpeed/GTmetrix/GTMetrixResultViewController.swift [hunk 33]
```diff
@@ -0,0 +1,173 @@
...
+                                            let path = (NSTemporaryDirectory() as NSString).appendingPathComponent(fileName)
+                                            do {
+                                                let url = URL(fileURLWithPath: path)
+                                                try data.write(to: url)
+                                                self?.urlPDF = url
+                                                self?.openPDF()
+                                            } catch {
+                                                print(error)
+                                            }
+                                        }
```
Comment (nedopaka) [inline PageSpeed/GTmetrix/GTMetrixResultViewController.swift:56]: @ilyageekhub indentation error. Each input parameter should start from new line in this case. Also, it should be intended with one additional level. See https://github.com/raywenderlich/swift-style-guide#function-calls.

File: PageSpeed/GTmetrix/GTMetrixResultViewController.swift [hunk 34]
```diff
@@ -0,0 +1,173 @@
...
+            }
```
Comment (nedopaka) [inline PageSpeed/GTmetrix/GTMetrixResultViewController.swift:57]: @ilyageekhub where is the closing parenthesis `)` for `GTMetrixResourceService`? Does this code work?

File: PageSpeed/GTmetrix/GTMetrixResultViewController.swift [hunk 35]
```diff
@@ -0,0 +1,173 @@
...
+    @IBOutlet private weak var labelFullyLoadedTime: UILabel!
+    @IBOutlet private weak var labelTotalPageSize: UILabel!
+    @IBOutlet private weak var labelRequests: UILabel!
+    @IBOutlet private weak var labelHTMLLoadTime: UILabel!
+    @IBOutlet private weak var labelPageLoadTime: UILabel!
+    @IBOutlet private weak var labelPageBytes: UILabel!
+    @IBOutlet private weak var labelHTMLBytes: UILabel!
+    @IBOutlet private weak var labelPageElements: UILabel!
+    @IBOutlet private weak var labelRumSpeedIndex: UILabel!
+    @IBOutlet private weak var buttonLoadPFD: UIButton!
```
Comment (nedopaka) [inline PageSpeed/GTmetrix/GTMetrixResultViewController.swift:28]: @ilyageekhub naming convention error (see p.4), use suffix to identify type, not prefix. E.g. `labelPageElements` should be **pageElementsLabel**.
>All IBOutlets, objects of classes and subclasses should have suffix according to their type (e.g. NewTaskViewController for UIViewController).

File: PageSpeed/GTmetrix/Services/GTMetrixResourceService.swift [hunk 36]
```diff
@@ -0,0 +1,57 @@
+//
+//  GTMetrixTestResourceService.swift
```
Comment (nedopaka) [inline PageSpeed/GTmetrix/Services/GTMetrixResourceService.swift:2]: @ilyageekhub this comment line should match file name.

File: PageSpeed/GTmetrix/Services/GTMetrixResourceService.swift [hunk 37]
```diff
@@ -0,0 +1,57 @@
...
+//  PageSpeed
+//
+//  Created by Admin on 10.03.2020.
```
Comment (nedopaka) [inline PageSpeed/GTmetrix/Services/GTMetrixResourceService.swift:5]: @ilyageekhub why you still use `Admin` as name? It's very confusing and uninformative.

File: PageSpeed/GTmetrix/Services/GTMetrixResourceService.swift [hunk 38]
```diff
@@ -0,0 +1,57 @@
...
+//  Copyright © 2020 Stanford University. All rights reserved.
+//
+
+import Moya
+
+enum TestResourse: String {
+    case screenshot, har, pagespeed, pagespeedFiles = "pagespeed-files", yslow, reportPdf = "report-pdf", video
+}
+class GTMetrixResourceService: Service {
```
Comment (nedopaka) [inline PageSpeed/GTmetrix/Services/GTMetrixResourceService.swift:14]: @ilyageekhub spacing error. Should be one blank line above.

File: PageSpeed/GTmetrix/Services/GTMetrixResultService.swift [hunk 39]
```diff
@@ -1,5 +1,5 @@
 //
-//  GTMTestService.swift
+//  GTMetrixTestService.swift
```
Comment (nedopaka) [inline PageSpeed/GTmetrix/Services/GTMetrixResultService.swift:2]: @ilyageekhub this comment line should match file name.

File: PageSpeed/GTmetrix/Services/GTMetrixService.swift [hunk 40]
```diff
@@ -1,5 +1,5 @@
 //
-//  GTMTestService.swift
+//  GTMetrixTestService.swift
```
Comment (nedopaka) [inline PageSpeed/GTmetrix/Services/GTMetrixService.swift:2]: @ilyageekhub this comment line should match file name.

File: PageSpeed/GTmetrix/Services/GTMetrixURLService.swift [hunk 41]
```diff
@@ -1,5 +1,5 @@
 //
-//  GTMTestURLService.swift
+//  GTMetrixTestURLService.swift
```
Comment (nedopaka) [inline PageSpeed/GTmetrix/Services/GTMetrixURLService.swift:2]: @ilyageekhub this comment line should match file name.

File: PageSpeed/Test/TestResultsViewController.swift [hunk 42]
```diff
@@ -13,10 +13,8 @@ class TestResultsViewController: UIViewController {
     // MARK: - Properties
     var url: String?
     var mobilePageSpeedResult, desktopPageSpeedResult: PageSpeedResponse?
-    var servicesArr: [
-    (id: String, name: String)
-    ]?
-    var gtmetrixResponse: GTMetrixTestStatusResponse?
+    var servicesArr: [(id: String, name: String)]?
+    var gTMetrixResponse: GTMetrixResponse?
```
Comment (nedopaka) [inline PageSpeed/Test/TestResultsViewController.swift:17]: @ilyageekhub once again, `gTMetrixResponse` is incorrect. Should be **gtMetrixResponse**, the same as at https://github.com/ANScorp/pagespeed-swift/pull/5/commits/d5b6264c7201999b5b73ab4b1a0ddc8a0ed097ea#diff-418cd2b81a677cb40ae2f9c2cbd9e28f.
See https://github.com/raywenderlich/swift-style-guide#naming.
>casing acronyms and initialisms uniformly up or down

File: PageSpeed/GTmetrix/Common/ServicesManager.swift [hunk 43]
```diff
@@ -8,7 +8,6 @@
 
 import Foundation
 
-// класс для хранения экземпляров сервисов, на время выполнения
 class ServicesManager {
```
Comment (nedopaka) [inline PageSpeed/GTmetrix/Common/ServicesManager.swift:11]: @ilyageekhub again, do we need this class? As we agreed it should be removed.

File: PageSpeed/GTmetrix/GTMetrixResultViewController.swift [hunk 44]
```diff
@@ -0,0 +1,173 @@
+//
+//  GTMetrixResaultViewController.swift
+//  PageSpeed
+//
+//  Created by Admin on 19.02.2020.
```
Comment (nedopaka) [inline PageSpeed/GTmetrix/GTMetrixResultViewController.swift:5]: @ilyageekhub why you still use `Admin` as name? It's very confusing and uninformative.

**Review by nedopaka — COMMENTED** [review]
@ilyageekhub fix errors mentioned below. Also, as for `ServicesManager` why are you using this class?
Again, why don't you **MARK** you code?

File: PageSpeed/GTmetrix/GTMetrixResultViewController.swift (hunk 33 above)
Comment (ilyageekhub) [reply PageSpeed/GTmetrix/GTMetrixResultViewController.swift:56]: Fix

File: PageSpeed/GTmetrix/GTMetrixResultViewController.swift (hunk 34 above)
Comment (ilyageekhub) [reply PageSpeed/GTmetrix/GTMetrixResultViewController.swift:57]: Yes. This code works.

File: PageSpeed/Test/TestResultsViewController.swift (hunk 42 above)
Comment (ilyageekhub) [reply PageSpeed/Test/TestResultsViewController.swift:17]: done

File: PageSpeed/GTmetrix/GTMetrixResultViewController.swift (hunk 35 above)
Comment (ilyageekhub) [reply PageSpeed/GTmetrix/GTMetrixResultViewController.swift:28]: fix

File: PageSpeed/GTmetrix/Services/GTMetrixService.swift (hunk 40 above)
Comment (ilyageekhub) [reply PageSpeed/GTmetrix/Services/GTMetrixService.swift:2]: fix

File: PageSpeed/GTmetrix/Services/GTMetrixURLService.swift (hunk 41 above)
Comment (ilyageekhub) [reply PageSpeed/GTmetrix/Services/GTMetrixURLService.swift:2]: fix

File: PageSpeed/GTmetrix/Common/ServicesManager.swift (hunk 43 above)
Comment (ilyageekhub) [reply PageSpeed/GTmetrix/Common/ServicesManager.swift:11]: This manager is necessary for keeping references of current active services. So its mandatory to keep it

File: PageSpeed/GTmetrix/GTMetrixResultViewController.swift (hunk 44 above)
Comment (ilyageekhub) [reply PageSpeed/GTmetrix/GTMetrixResultViewController.swift:5]: fix

**Pull request closed without merging**